mep/people/forms.py

```diff
@@ -110,22 +114,39 @@ class MemberSearchForm(FacetForm):
     }))
     membership_dates = RangeField(label='Membership Dates', required=False,
         widget=RangeWidget(attrs={'size': 4}))
+    birth_year = RangeField(required=False,
+        widget=RangeWidget(attrs={'size': 4}))
 
-    def set_membership_dates_placeholder(self, min_year, max_year):
-        '''Set the min, max, and placeholder values for
-        :class:`mep.common.forms.RangeWidget` associated with membership_dates.'''
-
-        start_widget, end_widget = \
-            self.fields['membership_dates'].widget.widgets
-
-        # set placeholders for widgets individually
-        start_widget.attrs['placeholder'] = min_year
-        end_widget.attrs['placeholder'] = max_year
-        # valid min and max for both via multiwidget
-        self.fields['membership_dates'].widget.attrs.update({
-            'min': min_year,
-            'max': max_year
-        })
+    def set_daterange_placeholders(self, min_max_conf):
+        '''Set the min, max, and placeholder values for all fields associated
+        with :class:`mep.common.forms.RangeWidget`.
+
+        :param min_max_conf: set of (min, max) tuples
+        :type min_max_conf: dict
+        :rtype: None
+        '''
+        for field, field_obj in self.fields.items():
```

Comment:
It seems inefficient to loop over all the fields in the form and check types, why is this not based on the min/max options that are passed in?

Reply:
This was the result of leaving the configuration on the form (which I had done because it seemed like the appropriate place to kept it because the form would be the class that knows about its configuration).

I'll address other comments and see if that doesn't let me swap this around.